UnicornEngine/WindowManager/include/unicorn/window_manager/Hub.hpp

```diff
@@ -0,0 +1,227 @@
+/*
+* Copyright (C) 2017 by Godlike
+* This code is licensed under the MIT license (MIT)
+* (http://opensource.org/licenses/MIT)
+*/
+
+#ifndef UNICORN_WINDOW_MANAGER_HUB_HPP
+#define UNICORN_WINDOW_MANAGER_HUB_HPP
+
+#include <unicorn/window_manager/Monitor.hpp>
+#include <unicorn/window_manager/Window.hpp>
+#include <unicorn/window_manager/WindowHint.hpp>
+
+#include <wink/signal.hpp>
+#include <wink/event_queue.hpp>
+
+#include <vulkan/vulkan.hpp>
+
+#include <map>
+#include <vector>
+
+namespace unicorn
+{
+namespace WindowManager
+{
+
+/** @brief  Provides interface to managing windows and monitors
+ *
+ *  Proxies calls to window manager adapter
+ */
+class Hub
+{
+public:
+    /** @brief  Constructs a hub object */
+    Hub();
+
+    Hub(const Hub& other) = delete;
+    Hub& operator=(const Hub& other) = delete;
+
+    Hub(Hub&& other) = delete;
+    Hub& operator=(Hub&& other) = delete;
+
+    /** @brief  Deconstructs a hub object
+     *
+     *  Also calls Deinit()
+     *
+     *  @sa Deinit()
+     */
+    ~Hub();
+
+    /** @brief  Creates a window object
+     *
+     *  @param  width           desired width
+     *  @param  height          desired height
+     *  @param  name            desired name
+     *  @param  pMonitor        pointer to a monitor where window
+     *                          should be created
+     *  @param  pSharedWindow   pointer to another window that should
+     *                          share all resources with created window
+     *
+     *  @return a pointer to newly created window object
+     *
+     *  @sa DestroyWindow()
+     */
+    Window* CreateWindow(int32_t width,
+        int32_t height,
+        const std::string& name,
+        Monitor* pMonitor = nullptr,
+        Window* pSharedWindow = nullptr);
+
+    /** @brief  Creates a vulkan surface that can be used with this window
+     *
+     *  @param[in]  window      window which should be associated with surface
+     *  @param[in]  instance    Vulkan instance
+     *  @param[in]  allocator   Vulkan allocator
+     *  @param[out] surface     variable that will hold created surface handle
+     *
+     *  @return Vulkan result flag
+     */
+    VkResult CreateVulkanSurfaceForWindow(const Window& window,
+        VkInstance instance,
```

Comment:
I am not sure about this one. There is no logic inside this method that utilizes c++ objects and if I understand correctly, the vk::Result can be initialized using VkResult. So I would just prefer to keep the API at the level it is operating on without creating extra objects.